src/firmware/shared/circular_buffer.c

```diff
@@ -0,0 +1,54 @@
+#include "firmware/shared/circular_buffer.h"
+
+void circular_buffer_init(struct circular_buffer *cbuff)
+{
+    cbuff->head   = 0;
+    cbuff->tail   = 0;
+    cbuff->isFull = false;
+    // Initialize buffer size to CIRCULAR_BUFFER_MAX_NUM_ELEMENTS
+    cbuff->bufferData = (float *)malloc(sizeof(float) * CIRCULAR_BUFFER_MAX_NUM_ELEMENTS);
```

Comment:
why not take this as an argument when initializing the buffer?

Reply:
Was wondering if it should in one of my earlier comments. 
It now does.